lib/nodeUtils.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function toJson(node) {
+    function deriveEndpointType(tags) {
+        const urlTypes = [];
+        Object.keys(tags).forEach((objectKey) => {
+            if (objectKey.indexOf('.url') !== -1 || objectKey.indexOf('.uri') !== -1) {
+                urlTypes.push(objectKey.substring(0, objectKey.length - 4).toUpperCase());
+            }
+        });
+
+        if (urlTypes.length > 0) {
+            return urlTypes[0];
+        }
+
+        return 'HTTP';
+    }
+
+    function deriveOperation(span) {
+        if (!span.getOperationName()) {
+            const httpMethod = span.getTags()['http.method'];
```

Comment:
Would it not be better to just `return span.getTags()['http.method'];` as operation name is undefined anyway, so falling through would have no benefit.